lib/codegen/meta/src/cdsl/instructions.rs

```diff
@@ -0,0 +1,112 @@
+use super::types::ValueType;
+
+pub struct InstructionList {
+    groups: Vec<InstructionGroup>,
+}
+
+impl InstructionList {
+    pub fn new() -> Self {
+        Self { groups: Vec::new() }
+    }
+
+    pub fn group(&mut self, name: &'static str, f: impl FnOnce(&mut InstructionGroup)) {
```

Comment:
Can you add a comment explaining how this would be used? I'm a little unclear, because I don't see an obviously corresponding construct in meta-python, and it's not immediately obvious on its own.